PowerEditor/src/WinControls/ProjectPanel/ProjectPanel.cpp

```diff
@@ -1208,10 +1787,246 @@ int FileRelocalizerDlg::doDialog(const TCHAR *fn, bool isRTL)
 	{
 		DLGTEMPLATE *pMyDlgTemplate = NULL;
 		HGLOBAL hMyDlgTemplate = makeRTLResource(IDD_FILERELOCALIZER_DIALOG, &pMyDlgTemplate);
-		int result = ::DialogBoxIndirectParam(_hInst, pMyDlgTemplate, _hParent,  dlgProc, (LPARAM)this);
+		int result = ::DialogBoxIndirectParam(_hInst, pMyDlgTemplate, _hParent,  dlgProc, reinterpret_cast<LPARAM>(this));
 		::GlobalFree(hMyDlgTemplate);
 		return result;
 	}
-	return ::DialogBoxParam(_hInst, MAKEINTRESOURCE(IDD_FILERELOCALIZER_DIALOG), _hParent,  dlgProc, (LPARAM)this);
+	return ::DialogBoxParam(_hInst, MAKEINTRESOURCE(IDD_FILERELOCALIZER_DIALOG), _hParent,  dlgProc, reinterpret_cast<LPARAM>(this));
+}
+
+INT_PTR CALLBACK FilterDlg::run_dlgProc(UINT Message, WPARAM wParam, LPARAM) 
+{
+	switch (Message)
+	{
+		case WM_INITDIALOG :
+		{
+			NppParameters *nppParams = NppParameters::getInstance();
+			FindHistory & findHistory = nppParams->getFindHistory();
+
+			fillComboHistory(IDD_PROJECTPANEL_FILTERS_COMBO, findHistory._findHistoryFilters);
+			goToCenter();
+			return TRUE;
+		}
+		case WM_COMMAND : 
+		{
+			switch (wParam)
+			{
+				case IDOK :
+				{
+					_filters.clear();
+					const int filterSize = 256;
+					TCHAR tfilters[filterSize + 1];
+					tfilters[filterSize] = '\0';
+					::GetDlgItemText(_hSelf, IDD_PROJECTPANEL_FILTERS_COMBO, tfilters, filterSize);

```

Comment:
You ***absolutely*** should check to see if `::GetDlgItemText` returned `0` before using `tfilters`.